pkg/devicetemplateconverter/devicetemplateconverter.go

```diff
@@ -45,12 +46,28 @@ var errNotFound = errors.DefineNotFound("converter", "converter `{id}` not found
 
 // New returns a new *DeviceTemplateConverter.
 func New(c *component.Component, conf *Config) (*DeviceTemplateConverter, error) {
+	// Overwrites the csv converter with a profile fetcher if address is given.
+	if conf.DeviceRepositoryAddress != "" {
+		devicetemplates.RegisterConverter(
+			devicetemplates.TTSCSV,
+			devicetemplates.NewCSV(&ProfileFetcher{
+				DeviceRepositoryAddress: conf.DeviceRepositoryAddress,
```

Comment:
You should use the peering functionality (allow `ProfileFetcher` to use `GetPeerConn`) instead. Carrying the address internally has its own issues (like missing `InsecureSkipVerify`).